wandb/__init__.py

```diff
@@ -687,8 +715,14 @@ def set_global_config(run):
         api = InternalApi()
         # let init_jupyter handle this itself
         if not in_jupyter and not api.api_key:
-            termlog("Not authenticated with Weights & Biases (https://wandb.com). To automatically "
-                    "save results, run \"wandb signup\" or \"wandb login\"")
+            if force:
+                termerror(
+                    "No credentials found.  Run \"wandb login\" or \"wandb off\" to disable wandb")
+            else:
+                termlog(
+                    "wandb isn't configured, run \"wandb sync\" from this directory to visualize metrics")
+                run.mode = "dryrun"
+                _init_headless(run, False)
```

Comment:
This logic should allow users to distribute code instrumented with wandb without requiring users to create accounts.